lib/handler/configurator/proxy.c

```diff
@@ -122,6 +123,9 @@ static SSL_CTX *create_ssl_ctx(void)
     options |= SSL_OP_NO_RENEGOTIATION;
 #endif
     SSL_CTX_set_options(ctx, options);
+    SSL_CTX_set_session_id_context(ctx, (const uint8_t*)"h2o", sizeof("h2o") - 1);
+    SSL_CTX_set_session_cache_mode(ctx, SSL_SESS_CACHE_CLIENT | SSL_SESS_CACHE_NO_INTERNAL_STORE);
+    SSL_CTX_sess_set_new_cb(ctx, h2o_socket_ssl_new_session_cb);
```

Comment:
These changes look good.

The only suggestion I have is to define a macro for the arguments of `SSL_CTX_set_session_id_context`, possibily declaring them in include/h2o/socket.h, and using them throughout this PR.